### PR TITLE
Adding Jenkins substitutions for log file names.

### LINKED ISSUE #18: build log upload: expanding build parameters doesn't work
I've tried giving the build log name of "build.$BUILD_NUMBER.log", and the upload failed.
For some reason the storage location can  expand parameters.
I guess this is something to do with the difference between the two.

### PR DESCRIPTION
**Pull request by agoulti**
Adding in the ability to do substitutions in Jenkins build log name, to allow build log names such as "build.$BUILD_NUMBER.log".

The implementation of doCheckLogName in StdoutUpload.java hints that this was intended behavior all along.

Fixes #18.

### CODE REVIEW

**Comment by agoulti** [comment]
Thank you for your review!

The refactoring was to allow a new test.

I have added a test to ensure that the log name expansion works properly (StdoutUploadTest -> doCheckLogNameExpansion). That test required MockUploadModule as well as some other functions implemented in AbstractUploadTest, so I have pulled them into a shared class to reduce repetition.

**Comment by mattmoor** [comment]
@oleg-nenashev What needs to happen to get this merged?  I originally wrote this (and several other plugins) but I left the Jenkins organization a while ago.  I would be happy to vouch for @agoulti as the new owner for the `google-*` plugins, but IDK what process is in place for that kind of ownership transfer.

File: src/main/java/com/google/jenkins/plugins/storage/StdoutUpload.java [hunk 1]
```diff
@@ -102,7 +103,10 @@ protected UploadSpec getInclusions(AbstractBuild<?, ?> build,
       OutputStream outputStream = null;
       try {
         FilePath logDir = new FilePath(build.getLogFile()).getParent();
-        FilePath logFile = new FilePath(logDir, getLogName());
+
+        String resolvedLogName =
+            Util.replaceMacro(getLogName(), build.getEnvironment(listener));
+        FilePath logFile = new FilePath(logDir, resolvedLogName);
```
Comment (jglick) [inline src/main/java/com/google/jenkins/plugins/storage/StdoutUpload.java]: Note that this sort of patch is unnecessary for Pipeline: the script can use GString interpolation without any special support from the plugin. [more](https://github.com/jenkinsci/pipeline-plugin/blob/master/DEVGUIDE.md#variable-substitutions)

**Comment by oleg-nenashev** [comment]
There is no special process for this case. If the current maintainer and
the proposed one are fine, we can transfer the ownership immediately. The
process applies only to plugins with unresponsive maintainers.

BR, Oleg

File: src/main/java/com/google/jenkins/plugins/storage/StdoutUpload.java (hunk 1 above)
Comment (agoulti) [reply src/main/java/com/google/jenkins/plugins/storage/StdoutUpload.java]: Thanks for pointing this out!
Currently the pipeline is unsupported, but I will make sure to hunt down these substitutions when we add pipeline compatibility (this plugin already has quite a number of other places where similar substitutions happen).

**Comment by agoulti** [comment]
@oleg-nenashev: Transferring immediately would be great!
I've put in a request here: https://groups.google.com/forum/#!topic/jenkinsci-dev/V6XoKTF8LYs

**Pull request merged**